### ISSUE TITLE
msfconsole -y parameter is not working

### ISSUE THREAD

**Issue by FireFart** [rapid7/metasploit-framework#10844]
I noticed this in https://github.com/rapid7/metasploit-framework/pull/10843

When running msfconsole with the -y parameter to supply a `database.yml` file, the file is not loaded correctly. In my case it should automatically connect to the database docker container, but it tries to connect to localhost instead. Localhost is set in the code as a default DB, but the parameter is not honoured.

Currently I get the error because some defaults are set anywhere in the code:
```
[-] ***Starting the Metasploit Framework console...-
[-] * WARNING: No database support: could not connect to server: Connection refused
	Is the server running on host "localhost" (127.0.0.1) and accepting
	TCP/IP connections on port 5432?
could not connect to server: Address not available
	Is the server running on host "localhost" (::1) and accepting
	TCP/IP connections on port 5432?
[-] ***
```

Expected behaviour: msfconsole connects to the database supplied in database.yml instead

**Comment by mkienow-r7** [comment]
I'm wondering if this is because `docker/msfconsole.rc` actually calls `db_connect` with `ENV['DATABASE_URL']`.

**Comment by FireFart** [comment]
@mkienow-r7 regarding your comment https://github.com/rapid7/metasploit-framework/pull/10843/files#r227136317:
If I remove the line, also no DB connection is established:
```text
[firefart@fedora metasploit-framework]$ ./docker/bin/msfconsole 
Starting metasploit-framework_db_1 ... done
[-] ***rTing the Metasploit Framework console...|
[-] * WARNING: No database support: ActiveRecord::ConnectionNotEstablished
[-] ***
                                                  
IIIIII    dTb.dTb        _.---._
  II     4'  v  'B   .'"".'/|\`.""'.
  II     6.     .P  :  .' / | \ `.  :
  II     'T;. .;P'  '.'  /  |  \  `.'
  II      'T; ;P'    `. /   |   \ .'
IIIIII     'YvP'       `-.__|__.-'

I love shells --egypt


       =[ metasploit v5.0.0-dev                           ]
+ -- --=[ 1826 exploits - 1033 auxiliary - 318 post       ]
+ -- --=[ 541 payloads - 44 encoders - 10 nops            ]
+ -- --=[ 2 evasion                                       ]
+ -- --=[ ** This is Metasploit 5 development branch **   ]

[*] Processing docker/msfconsole.rc for ERB directives.
[*] resource (docker/msfconsole.rc)> Ruby Code (165 bytes)
LHOST => 172.20.0.3
msf5 > db_status
[*] postgresql selected, no connection
msf5 > echo $DATABASE_URL
[*] exec: echo $DATABASE_URL

postgres://postgres@db:5432/msf_dev?pool=200&timeout=5
msf5 > ping -c 1 db
[*] exec: ping -c 1 db

PING db (172.20.0.2): 56 data bytes
64 bytes from 172.20.0.2: seq=0 ttl=64 time=0.070 ms

--- db ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 0.070/0.070/0.070 ms
msf5 > exit
```

**Comment by FireFart** [comment]
My modified `docker/bin/msfconsole` line:

```
docker-compose run --rm --service-ports -e MSF_UID=$(id -u) -e MSF_GID=$(id -g) ms ruby msfconsole -y docker/database.yml -r docker/msfconsole.rc "$PARAMS"
```

**Comment by busterb** [comment]
This was fixed in #10843 looks like.
